charts/vmclarity/templates/apiserver/deployment.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "vmclarity.apiserver.name" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "vmclarity.apiserver.labels.standard" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels: {{- include "vmclarity.apiserver.labels.matchLabels" . | nindent 6 }}
+  replicas: {{ .Values.apiserver.replicaCount }}
+  template:
+    metadata:
+      labels: {{- include "vmclarity.apiserver.labels.standard" . | nindent 8 }}
+    spec:
+      {{- if .Values.apiserver.affinity }}
+      affinity: {{- .Values.apiserver.affinity | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.apiserver.nodeSelector }}
+      nodeSelector: {{- .Values.apiserver.nodeSelector | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.apiserver.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.apiserver.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.apiserver.tolerations }}
+      tolerations: {{- .Values.apiserver.tolerations | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: apiserver
+          image: {{ include "vmclarity.images.image" ( dict "imageRoot" .Values.apiserver.image "global" .Values.global ) }}
```

Comment:
can you please explain the use of "dict" here?

Reply:
the `include` function takes two inputs, the name of the template to render, and a dictionary of the data to render the template with.

In this case the "vmclarity.images.image" template expects a dictionary with two top level keys "imageRoot" which is the image info in a standard structure and "global" which is the global part of values that might contain any overrides.